jupyterhub/auth.py

```diff
@@ -1520,7 +1577,13 @@ async def authenticate(self, handler, data):
         """Checks against a global password if it's been set. If not, allow any user/pass combo"""
         if self.password:
             if data['password'] == self.password:
-                return data['username']
+                # Anyone logging in with the standard password is *never* admin
+                return {"name": data['username'], "admin": False}
```

Comment:
I think I see a slight problem here: if a user knows the user password and an admin's username, they can still login as the admin user and _revoke_ the admin user's admin privileges. I think you also need to check `not in self.admin_users` before accepting the default password.

Reply:
Ah that's fair.